consul-on-ubuntu/azuredeploy.json

```diff
@@ -0,0 +1,327 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "appPrefix": {
+            "type": "string",
+            "metadata": {
+                "description": "Prefix for each component (VMs, networks, etc)"
+            }
+        },
+        "adminUsername": {
+            "type": "string",
+            "metadata": {
+                "description": "Admin username for VM"
+            }
+        },
+        "adminPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Admin password for VM"
+            }
+        },
+        "newStorageAccountName": {
+            "type": "string",
+            "metadata": {
+                "description": "Unique storage account name. Must be between 3 and 24 characters in length and use numbers and lower-case letters only."
+            }
+        },
+        "vmSize": {
+            "type": "string",
+            "defaultValue": "Standard_A1",
+            "allowedValues": [
+                "Standard_A1",
+                "Standard_A2",
+                "Standard_A3",
+                "Standard_D1",
+                "Standard_D2",
+                "Standard_D3"
+            ],
+            "metadata": {
+                "description": "Size of the Virtual Machine."
+            }
+        },
+        "atlasInfra": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of your infrastructure in Atlas (username/infraName)"
+            }
+        },
+        "atlasToken": {
+            "type": "string",
+            "metadata": {
+                "description": "Your access token from Atlas"
+            }
+        }
+    },
+    "variables": {
+        "numberOfInstances": 3,
+        "imagePublisher": "Canonical",
+        "imageOffer": "UbuntuServer",
+        "imageSKU": "14.04.3-LTS",
+        "customScriptFilePath": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/consul-on-ubuntu/install-consul.sh",
+        "customScriptCommandToExecute": "bash install-consul.sh",
+        "vnetIDRef": "[concat(parameters('appPrefix'),'_VNET')]",
+        "vnetID": "[resourceId('Microsoft.Network/virtualNetworks', variables('vnetIDRef'))]",
+        "addressPrefix": "10.0.0.0/16",
+        "subnet1Name": "Subnet-1",
+        "subnet1Prefix": "10.0.0.0/24",
+        "subnet1Ref": "[concat(variables('vnetID'),'/subnets/', variables('subnet1Name'))]",
+        "publicIPAddressType": "Dynamic",
+        "vmStorageAccountContainerName": "vhds"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Storage/storageAccounts",
+            "name": "[parameters('newStorageAccountName')]",
+            "apiVersion": "2015-05-01-preview",
```

Comment:
Use the GA API versions 2015-06-15 and also make apiVersion a variable and reuse across resources below